core/src/test/java/org/elasticsearch/index/query/GeoDistanceRangeQueryTests.java

```diff
@@ -0,0 +1,292 @@
+        if (randomBoolean()) {
+            int branch = randomInt(2);
+            switch (branch) {
+            case 0:
+                builder.from(fromValue);
+                break;
+            case 1:
+                builder.to(toValue);
+                break;
+            case 2:
+                builder.from(fromValue);
+                builder.to(toValue);
+                break;
+            }
+        } else {
+            int branch = randomInt(2);
+            switch (branch) {
+            case 0:
+                builder.from(fromValue + fromToUnits);
+                break;
+            case 1:
+                builder.to(toValue + fromToUnits);
+                break;
+            case 2:
+                builder.from(fromValue + fromToUnits);
+                builder.to(toValue + fromToUnits);
+                break;
+            }
+        }
+        if (randomBoolean()) {
+            builder.includeLower(randomBoolean());
+        }
+        if (randomBoolean()) {
+            builder.includeUpper(randomBoolean());
+        }
+        if (randomBoolean()) {
+            builder.geoDistance(randomFrom(GeoDistance.values()));
+        }
+        if (randomBoolean()) {
+            builder.unit(randomFrom(DistanceUnit.values()));
+        }
+        if (randomBoolean()) {
+            builder.optimizeBbox(randomFrom("none", "memory", "indexed"));
+        }
+        if (randomBoolean()) {
+            builder.coerce(randomBoolean());
+        }
+        if (randomBoolean()) {
+            builder.ignoreMalformed(randomBoolean());
+        }
+        return builder;
+    }
+
+    @Override
+    protected void doAssertLuceneQuery(GeoDistanceRangeQueryBuilder queryBuilder, Query query, QueryShardContext context)
+            throws IOException {
+        assertThat(query, instanceOf(GeoDistanceRangeQuery.class));
+        GeoDistanceRangeQuery geoQuery = (GeoDistanceRangeQuery) query;
+        assertThat(geoQuery.fieldName(), equalTo(queryBuilder.fieldName()));
+        if (queryBuilder.lat() != null) {
+            assertThat(geoQuery.lat(), equalTo(queryBuilder.lat()));
+        }
+        if (queryBuilder.lon() != null) {
+            assertThat(geoQuery.lon(), equalTo(queryBuilder.lon()));
+        }
+        assertThat(geoQuery.geoDistance(), equalTo(queryBuilder.geoDistance()));
+        if (queryBuilder.from() != null && queryBuilder.from() instanceof Number) {
+            double fromValue = ((Number) queryBuilder.from()).doubleValue();
+            assertThat(geoQuery.minInclusiveDistance(), closeTo(fromValue, fromValue / 1000));
+        }
+        if (queryBuilder.to() != null && queryBuilder.to() instanceof Number) {
+            double toValue = ((Number) queryBuilder.to()).doubleValue();
+            assertThat(geoQuery.maxInclusiveDistance(), closeTo(toValue, toValue / 1000));
+        }
+    }
+
+    public String randomGeohash(int minPrecision, int maxPrecision) {
```

Comment:
I actually want to merge the GeohashCellQuery PR first which has this method on the BaseQueryTestCase and then use that method here. I'll make the method static in that PR though